nuxeo-core/nuxeo-core-cache/src/test/java/org/nuxeo/transientstore/test/TransientStorageComplianceFixture.java

```diff
@@ -338,4 +343,36 @@ public void verifyDeleteAfterUseGC() throws Exception {
         assertFalse(ts.exists("XXX"));
     }
 
+    @Test
+    public void verifyStorePathCanBeSpecified() throws Exception {
+        Framework.getProperties().setProperty("nuxeo.data.dir", Environment.getDefault().getData().getAbsolutePath());
+
+        TransientStoreService tss = Framework.getService(TransientStoreService.class);
+
+        // Verify default behavior (store cache dir is in ${nuxeo.data.dir}/transientstores/{name}
+        AbstractTransientStore ts = (AbstractTransientStore) tss.getStore("microStore");
+        assertEquals(Framework.expandVars("${nuxeo.data.dir}/transientstores/microStore"),
+                ts.getCacheDir().getAbsolutePath());
+
+        // Verify when a path is given
+        deployInlineStore("testPath", "${nuxeo.data.dir}/test");
+        ts = (SimpleTransientStore) tss.getStore("testPath");
+        assertEquals(new File(Environment.getDefault().getData(), "test").getAbsolutePath(),
+                ts.getCacheDir().getAbsolutePath());
+    }
+
+    // Deploys a store given a name and a path. This uses an inline componentRef to be able to expand vars in
+    // contribution.
+    private void deployInlineStore(String name, String path) throws IOException {
```

Comment:
I would have deployed a regular XML file containing framework properties for name and path, and deploy-undeploy it from inside the test using the harness.